### ISSUE TITLE
PowerPC InvalidInstruction

### ISSUE THREAD

**Issue by slimjim2234** [icicle-emu/icicle-emu#57]
I'd like to preface this with gratitude, thank you for developing an emulator that makes harnessing binaries sourced from uncommon architectures an achievable task.

That said, if you could shine some light in an instruction decoding error I would greatly appreciate it.

### Problem 
Add `0xc1000000 [7c 60 25 26]    "mfsrin r3,r4";` to `icicle-test/tests/powerpc.ins`
Run `cargo run powerpc powerpc` in the `icicle-test` folder

### Result
`[./tests/powerpc.ins:9] decoded instruction count 0 != expected 1 group Err(InvalidInstruction)`

### Details
I've traced it to the following function in `sleigh/sleigh-runtime/src/lifter.rs` where a match on `get_runtime_value` returns a non-constant offset for the varnode in question that is logically invalid for this function to continue successfully.  The function is returned to ` SemanticAction::LoadRegister` in `build_subtable` and is then caught by the block lifter on line 98 of `icicle-cpu/src/lifter/mod.rs`.
Unfortunately, I don't understand the purpose of this function enough to determine the correct outcome.

```
fn resolve_dynamic_varnode(&mut self, pointer: &Value, size: u16) -> Result<VarNode> {
        let value = self.resolve_value(*pointer)?;
        match self.get_runtime_value(value)? {
            pcode::Value::Const(offset, base_size) => {
                Ok(VarNode::register(offset as u32, base_size as u16).slice(0, size))
            }
            pcode::Value::Var(_) => Err(Error::VarNodeOffsetIsNotConstant),
        }
    }
```
At the moment of error, the value of `value` at the match is
`{offset:16, base_size:16, size:4, is_tmp:true}`

### Concluding thoughts
The slaspec details the instruction on line 2428 in `/opt/ghidra/Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc`.  The instruction uses special purpose segment registers sr0-sr15 used in address translation.
Knowing this, my best guess is icicle is treating segment registers as temporary or non-constant.  Need assistance validating and/or solving.

**Comment by mchesser** [comment]
The issue is related to the way Icicle translates dynamically computed registers (SLEIGH offsets) to the concrete register indices used by the emulator. That function attempts to handle that translation when the offset is computed from a disassembly time constant.

(i.e., we translate `register[0x1000]` to something like: `$r0`)

I had assumed that dynamically computed register offsets would always be resolved at disassembly time (based on '7.7.1.6. Address-of Operator' in the SLEIGH reference manual), but that appears to not be the case for PowerPC which essentially does `register[r4 >> 28]`.

This should be fairly straightforward to for me to support -- though I will caution that SLEIGH doesn't actually encode semantics for 
using the segment registers.

**Comment by slimjim2234** [comment]
Much appreciated, makes sense.
Yea, I assumed the segment registers won't play an accurate role.  Not concerned about it.
